crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1001 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Write},
+    result::Result,
+};
+
+use bevy_utils::{
+    petgraph::{algo::tarjan_scc, prelude::*},
+    thiserror::Error,
+    tracing::{error, warn},
+    HashMap, HashSet,
+};
+
+use fixedbitset::FixedBitSet;
+
+use crate::component::ComponentId;
+use crate::{
+    self as bevy_ecs,
+    schedule_v3::*,
+    system::{BoxedSystem, Resource},
+    world::World,
+};
+
+/// Stores [`ScheduleLabel`]-[`Schedule`] pairs.
+#[derive(Default, Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Option<Schedule>>,
+}
+
+impl Schedules {
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::new(),
+        }
+    }
+
+    /// Insert a new labeled schedule into the map.
+    ///
+    /// # Errors
+    ///
+    /// TODO
+    pub fn insert(
+        &mut self,
+        label: impl ScheduleLabel,
+        schedule: Schedule,
+    ) -> Result<(), InsertionError> {
+        let label = label.dyn_clone();
+        if self.inner.contains_key(&label) {
+            return Err(InsertionError::AlreadyExists(label));
+        }
+        self.inner.insert(label, Some(schedule));
+        Ok(())
+    }
+
+    /// Returns the schedule corresponding to the label.
+    ///
+    /// # Errors
+    ///
+    /// TODO
+    pub(crate) fn check_out(
```

Comment:
These seem genuinely useful outside of this crate: I would make this and its sibling pub.

Reply:
Hmm, so I made these private after realizing I had no mechanism to stop someone from checking out a schedule, then checking in a different one under the same label.

Fine with making them `pub`, but this was similar to how we don't actually want people re-configuring the `Core` system sets (but don't have a nice way to stop that).

Reply:
That seems like a feature to me, rather than a bug. This would be a very powerful tool for games with modding.

Reply:
Okay, in that case, I've replaced these and am just passing through the standard `insert`, `remove`, `get`, and `get_mut` API.